pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -129,10 +130,22 @@ func (s *Server) initDNSCerts(hostname, customHost, namespace string) error {
 
 	var certChain, keyPEM, caBundle []byte
 	var err error
-	if features.PilotCertProvider == constants.CertProviderKubernetes {
+	if features.PilotCertProvider == constants.CertProviderKubernetes && signerName != "" && s.RA != nil {
+		log.Infof("Generating K8S-signed cert for %v using signer %v", names, signerName)
+		caCertFile := path.Join(ra.DefaultExtCACertDir, constants.CACertNamespaceConfigMapDataName)
```

Comment:
Why do we use a different ca cert file here?

Reply:
If we are using RA, we should use the root cert from RA.

Reply:
To clarify: If we use ANY external K8s signer, this means that the CA cert file will be different. 
Note: We still need a good way to propagate the external ca to all the workloads so that they can connect with Istiod.